packages/jest-cli/src/cli/args.js

```diff
@@ -36,6 +36,14 @@ export const check = (argv: Argv) => {
     );
   }
 
+  if (argv.hasOwnProperty('maxWorkers') && argv.maxWorkers === undefined) {
+    throw new Error(
+      'The --maxWorkers option requires a number to be specified.\n' +
+        'Example usage: jest --maxWorkers 2\n' +
+        'Or did you mean --watch ?',
```

Comment:
nit: remove the space before question mark.

Also, how about adjusting this to most common mistake:
```diff
- The --maxWorkers option
+ The --maxWorkers (-w) option
```